From f65d7a1fed36aad0ac22c7582c1bf90a6aba9024 Mon Sep 17 00:00:00 2001
From: Dinolek <git@dinolek.me>
Date: Mon, 7 Jul 2025 14:19:16 +0200
Subject: [PATCH 043/115] arm64: dts: mediatek: mt6785: Add pwm support

MT6785 SoC has 4 PWMs. Add the pwm node in order to support them.

Signed-off-by: Dinolek <git@dinolek.me>
---
 arch/arm64/boot/dts/mediatek/mt6785.dtsi | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/arch/arm64/boot/dts/mediatek/mt6785.dtsi b/arch/arm64/boot/dts/mediatek/mt6785.dtsi
index 0b0b20383..e2ff4627c 100644
--- a/arch/arm64/boot/dts/mediatek/mt6785.dtsi
+++ b/arch/arm64/boot/dts/mediatek/mt6785.dtsi
@@ -536,6 +536,21 @@ i2c6: i2c@11005000 {
 			status = "disabled";
 		};
 
+		pwm0: pwm@11006000 {
+			compatible = "mediatek,mt6785-pwm", "mediatek,mt8183-pwm";
+			reg = <0 0x11006000 0 0x1000>;
+			#pwm-cells = <2>;
+			interrupts = <GIC_SPI 200 IRQ_TYPE_LEVEL_LOW>;
+			clocks = <&infracfg CLK_INFRA_PWM>,
+				 <&infracfg CLK_INFRA_PWM_HCLK>,
+				 <&infracfg CLK_INFRA_PWM1>,
+				 <&infracfg CLK_INFRA_PWM2>,
+				 <&infracfg CLK_INFRA_PWM3>,
+				 <&infracfg CLK_INFRA_PWM4>;
+			clock-names = "top", "main", "pwm1", "pwm2", "pwm3", "pwm4";
+			status = "disabled";
+		};
+
 		i2c0: i2c@11007000 {
 			compatible = "mediatek,mt6785-i2c", "mediatek,mt8186-i2c";
 			reg = <0 0x11007000 0 0x1000>, <0 0x11000080 0 0x80>;
-- 
2.56.0

